static/js/panels.js

```diff
@@ -83,7 +84,19 @@ export function initialize() {
     if (page_params.insecure_desktop_app) {
         open($("[data-process='insecure-desktop-app']"));
     } else if (page_params.server_needs_upgrade) {
-        open($("[data-process='server-needs-upgrade']"));
+        const last_notification_dismissal_time =
+            localstorage.supported() && ls.get("lastNotificationDismissalTime");
+
+        const upgrade_nag_dismissal_duration =
+            last_notification_dismissal_time &&
+            addDays(new Date(last_notification_dismissal_time), 7);
+
+        if (
+            !last_notification_dismissal_time ||
+            !(last_notification_dismissal_time < upgrade_nag_dismissal_duration)
+        ) {
```

Comment:
I don't understand this logic, and suspect is it is wrong.  We want to be comparing `now` to `last_notification_dismissal_time + 7 days`.  Whereas you seem to be checking something like `x + 7 days > x`.

In addition to fixing that, can you extract this test logic to a function, and write some node tests for it in a new `frontend_tests/node_tests/panels.js`?

Reply:
> I don't understand this logic, and suspect is it is wrong. We want to be comparing now to last_notification_dismissal_time + 7 days. Whereas you seem to be checking something like x + 7 days > x.

Yeah, you are right, What was I thinking! 🤦

Reply:
> In addition to fixing that, can you extract this test logic to a function, and write some node tests for it in a new frontend_tests/node_tests/panels.js?

Should I also add tests for other functions like `should_show_notifications` and `check_profile_incomplete`?

Reply:
Ideally, but if it proves time-consuming, feel free to just do that in a follow-up PR.

Reply:
(This PR is a release-blocker, so I wouldn't add an extra day or two to when it's finished to add tests for code that we've already hand-tested.)